### PR TITLE
add golangci-lint to make test

### PR DESCRIPTION
**Pull request by jrobsonchase**
## What
*Describe what the change is solving*

## How
*Describe the solution*

## Breaking Changes
*Are there any breaking changes in this PR?*

### CODE REVIEW

**Comment by jrobsonchase** [comment]
Current dependencies on/for this PR:
* main
  * **PR #227** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/227" target="_blank">Graphite</a> 
    * **PR #229** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/229" target="_blank">Graphite</a> 
      * **PR #230** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/230" target="_blank">Graphite</a> 
        * **PR #231** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/231" target="_blank">Graphite</a>  👈
    * **PR #228** <a href="https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/228" target="_blank">Graphite</a> 

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/ngrok/kubernetes-ingress-controller/231?utm_source=stack-comment).

**Comment by jrobsonchase** [comment]
Welp, no clue why the build job won't run :shrug:

File: .github/workflows/ci.yaml [hunk 1]
```diff
@@ -125,10 +126,8 @@ jobs:
         kubectl get nodes
         kubectl get pods -A
 
-    - name: Set up Go
-      uses: actions/setup-go@v3
-      with:
-        go-version: ${{ env.GO_VERSION }}
+    - name: Run direnv
+      uses: jrobsonchase/direnv-action@0.7
```
Comment (bobzilladev) [inline .github/workflows/ci.yaml:130]:
```suggestion
      uses: jrobsonchase/direnv-action@v0.7
```

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -113,7 +113,8 @@ jobs:
     - kubebuilder-diff
     if: |
       (needs.changes.outputs.go == 'true') ||
-      (needs.changes.outputs.charts == 'true')
+      (needs.changes.outputs.charts == 'true') ||
+      ('true' == 'true')
```
Comment (bobzilladev) [inline .github/workflows/ci.yaml:117]: Don't forget to undo this bit

File: .github/workflows/ci.yaml (hunk 1 above)
Comment (jrobsonchase) [reply .github/workflows/ci.yaml:130]: Whoops, should've copy-pasted :roll_eyes:

**Comment by jrobsonchase** [comment]
Hm, still not running :think

**Comment by bobzilladev** [comment]
Looks like it eventually ran, seeing the linting lines: https://github.com/ngrok/kubernetes-ingress-controller/actions/runs/5060657393/jobs/9083840896?pr=231

**Comment by jonstacks** [comment]
We pulled in some changes from kubebuilder's Makefile template and now have make `make lint` and `make lint-fix` available, just a heads up. I think we can turn it on once we get some of the existing lint issues fixed.

**Comment by jonstacks** [comment]
@jrobsonchase, I believe we covered this in https://github.com/ngrok/kubernetes-ingress-controller/pull/385 and should be good to go now. Its under a separate make target, `lint` and `lint-fix. Please feel free to re-open if I got it wrong

**Pull request closed without merging**